### DISCUSSION TITLE
Unresolvable dependency

### DISCUSSION THREAD

**Discussion by robinmaisch** [Fraunhofer-AISEC/cpg#1339] [Q&A] [answered]
I'm trying to use CPG as a library in a Maven project. I have succeded in supplying the Eclipse CDT dependency manually, but I'm out of luck getting the cpg-neo4j module (and the modules depending on it, e.g. cpg-console) to work because they have a dependency to the aggregate module `org.neo4j:neo4j-ogm`, which is not available as a JAR file in any repositories and also not buildable locally because the neo4j-ogm project is configured to only build a POM file for it.

Error message from IntelliJ:

```Could not find artifact org.neo4j:neo4j-ogm:jar:4.0.6 in central (https://repo.maven.apache.org/maven2)```

from cpg/cpg-neo4j/build.gradle.kts:

```kotlin
dependencies {
	//neo4j
	api(libs.bundles.neo4j)
	//...
}
``` 

from cpg/gradle/libs.versions.toml:
```
[bundles]
# ...
neo4j = ["neo4j-ogm-core", "neo4j-ogm", "neo4j-ogm-bolt"]
```

The [reference manual for neo4j-ogm](https://neo4j.com/docs/ogm-manual/current/reference/#reference:getting-started:dependency-management:gradle) states that in place of the `neo4j-ogm` module itself, the `neo4j-ogm-core` and `neo4j-ogm-bolt-driver` modules should be used as dependencies instead, so it seems the aggregate module would not be needed to be mentioned in the dependencies.

Am I wrong to think that the dependency to `neo4j-ogm` should be removed? Thanks for any hints. Cheers!

**Comment by KuechA** [comment]
Hi @robinmaisch,
I'm wondering why you need the `cpg-console` and `cpg-neo4j` subprojects if you want to use the CPG as a library in another project. These should not really be necessary and instead, you could configure only `cpg-core` and the required language-modules (and, if required the `cpg-analysis` module). This should avoid your dependency-problem.

**Reply by oxisto** [reply]
We are looking into why the `neo4j-ogm` bundle is there in the first place, it should probably not be there.

**Reply by robinmaisch** [reply]
Thank you @oxisto @KuechA! In case you're still wondering, I was trying to replicate some of the console functions inside my project and wanted to step through the code in order to understand better how to go about that, especially since I am not very experienced in Kotlin.

**Comment by oxisto** [comment, marked as answer]
I removed the `neo4j-ogm` bundle dependency in https://github.com/Fraunhofer-AISEC/cpg/pull/1340/files. We will release a `v8` version (with some breaking changes) soon.
